torch/_inductor/scheduler.py

```diff
@@ -638,6 +653,7 @@ def get_ranges(self):
         return self._sizes
 
     def is_reduction(self):
+        assert isinstance(self.node, ir.ComputedBuffer)
```

Comment:
I actually feel that it's better to keep this assert. IMO this piece of code is a bit hard to follow as node type is unclear. Also if self.node is not ir.ComputedBuffer the next statement will also fail, and adding an assertion message here can give clear error messages.

Reply:
This boils down to whether we are thinking of readability in code review or readability in the IDE, I guess :) Personally I think statically encoding the type information makes for clearer code because it's more clear why we expect `self.node` to be a certain type. But I guess this approach is clean enough...

Reply:
I see your point. Tried it, it caused more errors because `ForEachSchedulerNode` and `FusedSchedulerNode` don't have node defined.  I feel it brings more troubles than gains..

Reply:
Ah fair enough. Thanks for trying it!